sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/GpuBroadcastHashJoinExec.scala

```diff
@@ -42,16 +43,60 @@ class GpuBroadcastHashJoinMeta(
     join.leftKeys.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
   val rightKeys: Seq[BaseExprMeta[_]] =
     join.rightKeys.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
-  val condition: Option[BaseExprMeta[_]] =
+  val conditionMeta: Option[BaseExprMeta[_]] =
     join.condition.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
 
+  def isAstCondition(): Boolean = {
+    join.condition.isDefined && conditionMeta.forall(_.canThisBeAst)
+  }
+
   override val namedChildExprs: Map[String, Seq[BaseExprMeta[_]]] =
-    JoinTypeChecks.equiJoinMeta(leftKeys, rightKeys, condition)
+    JoinTypeChecks.equiJoinMeta(leftKeys, rightKeys, conditionMeta)
 
-  override val childExprs: Seq[BaseExprMeta[_]] = leftKeys ++ rightKeys ++ condition
+  override val childExprs: Seq[BaseExprMeta[_]] = leftKeys ++ rightKeys ++ conditionMeta
 
   override def tagPlanForGpu(): Unit = {
-    GpuHashJoin.tagJoin(this, join.joinType, join.leftKeys, join.rightKeys, join.condition)
+    //GpuHashJoin.tagJoin(this, join.joinType, join.leftKeys, join.rightKeys, join.condition)
+    val joinType = join.joinType
+    val keyDataTypes = (leftKeys ++ rightKeys).map(_.dataType)
+    val gpuBuildSide = GpuJoinUtils.getGpuBuildSide(join.buildSide)
+
+    def unSupportNonEqualCondition(): Unit = if (join.condition.isDefined) {
+      this.willNotWorkOnGpu(s"$joinType joins currently do not support conditions")
+    }
+    def unSupportNonEqualNonAst(): Unit = if (join.condition.isDefined) {
+      conditionMeta.foreach(requireAstForGpuOn)
+    }
+    def unSupportStructKeys(): Unit = if (keyDataTypes.exists(_.isInstanceOf[StructType])) {
+      this.willNotWorkOnGpu(s"$joinType joins currently do not support with struct keys")
+    }
+
+    JoinTypeChecks.tagForGpu(joinType, this)
+    joinType match {
+      case _: InnerLike =>
+      case RightOuter | LeftOuter | LeftSemi | LeftAnti =>
+        unSupportNonEqualNonAst()
+      case FullOuter =>
+        unSupportNonEqualCondition()
+        // FullOuter join cannot support with struct keys as two issues below
+        //  * https://github.com/NVIDIA/spark-rapids/issues/2126
+        //  * https://github.com/rapidsai/cudf/issues/7947
+        unSupportStructKeys()
+      case _ =>
+        this.willNotWorkOnGpu(s"$joinType currently is not supported")
+    }
+    // If there is an AST condition, we need to make sure the build side works for
+    // GpuBroadcastNestedLoopJoin
+    if (isAstCondition()) {
+      join.joinType match {
```

Comment:
nit:
```suggestion
      joinType match {
```

there are other uses of `join.joinType` but you defined `joinType` so I'd make them all consistent.

Reply:
Yeah this was another minor change to minimize diffs with GpuHashJoin.tagJoin.  I will make them consistent.